### PR TITLE
[WIP] use ONEMKL_LIBRARIES for mkl libs in torch

### PR DESCRIPTION
**Pull request by min-jean-cho**
**Background**
IPEX-GPU build fails on Windows after oneAPI 2024.1 uplift:
`LINK : fatal error LNK1104: cannot open file 'mkl_blacs_ilp64.lib'
`

**Root cause**
- **from oneMKL side**:
oneMKL exposes internal MKLConfig.cmake variable [MKL_LIBRARIES](https://github.com/oneapi-src/oneMKL/blob/develop/cmake/mkl/MKLConfig.cmake#L108) that's not supposed to be an output variable.
`MKL_LIBRARIES` in oneMKL 2024.1 contains unnecessary and incorrect MKL libs (`mkl_scalapack_ilp64.lib` `mkl_cdft_core.lib` `mkl_blacs_ilp64.lib`). In the future, they plan to hide the internal variable, `MKL_LIBRARIES`. 

- **from PyTorch side**:
Hence when oneMKL 2024.1 env is activated, PyTorch `MKL_LIBRARIES` from pytorch's own find mkl cmakes with `find_package(MKL ...)`  gets overwritten by oneMKL `MKL_LIBRARIES` from MKLConfig.cmake. 

**Solution**
Renaming pytorch's `MKL_LIBRARIES` naming to `ONEMKL_LIBRARIES` will resolve the root cause from pytorch side.
And it's better not to use `MKL_LIBRARIES` naming that may conflict with oneMKL's internal cmake variable in long term.

cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @yanbing-j @Guobing-Chen @Xia-Weiwen @snadampal

### CODE REVIEW

**Comment by min-jean-cho** [comment]
@guangyey , @CuiYifeng , please help review. Thanks.

**Review by guangyey — APPROVED** [review]

**Review by CuiYifeng — APPROVED** [review]

File: cmake/public/mkl.cmake
Comment (EikanWang) [inline cmake/public/mkl.cmake:1]: I think we just need to refine the `find_package` a little bit.

`find_package(MKL QUIET MODULE)`

Comment (EikanWang) [reply cmake/public/mkl.cmake:1]: [`find_package` ](https://cmake.org/cmake/help/latest/command/find_package.html#search-modes) FYI

**Comment by min-jean-cho** [comment]
Re-open when ready.

**Pull request closed without merging**